pkg/api/validate/dynamic/dynamic.go

```diff
@@ -80,46 +79,59 @@ func (dv *dynamic) ValidateVnet(ctx context.Context, location string, subnets []
 	}
 
 	// each subnet is threated individually as it would be from the different vnet
+	// During cluster runtime worker profile gets enriched and contains multiple
+	// duplicate values for multiple worker pools. We care only about
+	// unique subnet value in the functions below.
+	subnets = uniqueSubnetSlice(subnets)
+
+	// get unique vnets from subnets
+	vnets := make(map[string]azure.Resource)
 	for _, s := range subnets {
-		err := dv.validatePermissions(ctx, s)
+		vnetID, _, err := subnet.Split(s.ID)
 		if err != nil {
 			return err
 		}
 
-		err = dv.validateRouteTablePermissions(ctx, s)
+		vnetr, err := azure.ParseResourceID(vnetID)
 		if err != nil {
 			return err
 		}
+		vnets[vnetID] = vnetr
```

Comment:
```suggestion
		vnets[strings.ToLower(vnetID)] = vnetr
```

^ would be a little stronger
(also do the same in uniqueSubnetSlice)